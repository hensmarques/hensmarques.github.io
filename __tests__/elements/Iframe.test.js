import React from 'react';
import ShiftIframe from '../../src/elements/Iframe';

describe('<ShiftIframe/>', () => {
  it('renders with default props and matches the snapshot', () => {
    const wrapper = shallow(<ShiftIframe />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the props and children correctly', () => {
    const mockProps = {
      className: 'mockClass',
      width: '50%',
      height: '50%',
      allowFullScreen: true,
      src: 'mockSrc',
      id: 'mockId'
    };
    const wrapper = shallow(<ShiftIframe {...mockProps}>mockChild</ShiftIframe>);
    expect(wrapper.props().className).toBe(mockProps.className);
    expect(wrapper.props().width).toBe(mockProps.width);
    expect(wrapper.props().height).toBe(mockProps.height);
    expect(wrapper.props().allowFullScreen).toBe(mockProps.allowFullScreen);
    expect(wrapper.props().src).toBe(mockProps.src);
    expect(wrapper.props().id).toBe(mockProps.id);
    expect(wrapper.text()).toBe('mockChild');
  });
});
