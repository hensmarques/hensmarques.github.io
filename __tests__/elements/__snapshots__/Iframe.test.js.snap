// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShiftIframe/> renders with default props and matches the snapshot 1`] = `
<iframe
  allowFullScreen={false}
  frameBorder="0"
  height="100%"
  width="100%"
/>
`;
